interpreter/interpreter.go

```diff
@@ -17,12 +19,16 @@ type Interpreter struct {
 	polyTypes   map[semantic.Node]semantic.PolyType
 	sideEffects []SideEffect // a list of the side effects occurred during the last call to `Eval`.
 	pkg         string
+	deps        dependencies2.Interface
+	ctx         context2.Context
 }
 
-func NewInterpreter() *Interpreter {
+func NewInterpreter(ctx context2.Context, deps dependencies2.Interface) *Interpreter {
```

Comment:
Similar suggestion here to not store the context but rather to pass it in for calls to `Eval`.